### ISSUE TITLE
Files resolved via webpack configured modules are categorised as "undetermined"

### ISSUE THREAD

**Issue by BernieSumption** [sverweij/dependency-cruiser#51]
First off, thanks for the excellent work on this project. I've been waiting for #4 to be merged so I can use it in a webpack project, and it's working very nicely. I've found one issue with the new webpack resolution.

### Expected behaviour

When a file contains an import from e.g. "shared/foo" which webpack resolves to a file in the local source tree (say ./src/shared/foo.ts), then dependency cruiser should classify the dependency as "local".

### Current Behavior

The dependency is correctly mapped - the source and destination files paths are correctly resolved - but the dependency is classified as "undetermined".

### Steps to Reproduce (for bugs)

Key files excerpted below, hopefully this is enough detail but let me know if you'd like me to make a working minimal repro.

**src/A.ts:**
```ts
import foo from "shared/B"
```

**src/shared/B.ts**
```ts
export default function X() {}
```

**webpack.config.js**
```js
module.exports = {
  context: path.resolve(__dirname, "src"),
  resolve: {
    // treat all folders in src as modules
    modules: [path.resolve(__dirname, "src"), "node_modules"],
    extensions: [".ts", ".tsx", ".js", ".json"]
  }
}
```

**.dependency-cruiser.json**
```js
// in "rules"
    {
      "name": "no-undetermined",
      "severity": "error",
      "comment": "Don't allow undetermined dependencies",
      "from": {
        "path": ".*"
      },
      "to": {
        "dependencyTypes": ["undetermined"]
      }
    }
```

**CLI output**
```
depcruise --validate .dependency-cruiser.json src/A.ts

error no-undetermined: src/A.ts → src/screens/B.ts
```

### Your Environment

* Version used: 4.1.1
* Node version: v8.11.2
* Operating System and version: Mac OS 10.13.5
* Link to your project: private, but can provide a public minimal reproduction if required

**Comment by BernieSumption** [comment]
Worth noting - there is a relatively simple workaround of using "pathNot" to exclude any dependencies that point to the src folder.

```js
// in "rules"
    {
      "name": "no-undetermined",
      "severity": "error",
      "comment": "Don't allow undetermined dependencies",
      "from": {
        "path": ".*"
      },
      "to": {
        "pathNot": "^src/",
        "dependencyTypes": ["undetermined"]
      }
    }
```

**Comment by sverweij** [comment]
Hi @BernieSumption, thanks for reporting this. And for the level of detail in your report!

dependency-cruiser has the dependency type _undetermined_ as a fallback for when all else fails. In this case it should have classified the `shared/B` as something _similar_ to node_modules; webpack's  `modules` attribute is meant to do just that. Sorting your local 'modules' to _exactly_ the same as `node_modules` is not going to cut it, though. A lot of webpack configs I have seen use the `modules` attribute to solve `'../'.repeat(N)`-syndrome.

So my proposal is this:
- all sources in `node_modules` and/ or the ones defined in the `resolve.module` array will get an additional dependency type: `module` (exact name to be decided), 
- This dependency type will come _on top_ of the `npm*` dependency types for node_modules.
- Anything that's a `module` but not in a `node_modules` folder will also get be classified as `local` dependency type.

(Simplified) dependency-cruiser raw object response snippet:

```javascript
...
        {
            "source": "src/A.ts",
            "dependencies": [
                {
                    "module": "shared/B",
                    "resolved": "src/shared/B.ts",
                    "dependencyTypes": [
                        "module", //new
                        "local" // in stead of the erroneous 'undetermined'
                    ]
                },
                {
                    "module": "hidash",
                    "resolved": "node_modules/hidash/dist/index.js",
                    "dependencyTypes": [
                        "module", // new
                        "npm"
                    ]
                }
        },
...
```

> (B.t.w. already on the backlog was to do something similar for the `alias` section - any alias currently only gets the `alias` dependency type assigned, while they might also be `local` or even one of the npm deps. I'll solve that separately, though.

**Comment by sverweij** [comment]
@BernieSumption you can try it out by installing the beta version from npm (`npm i dependency-cruiser@4.2.0-beta-0`), which now sorts local 'modules' into `localmodule`

Let me know what you think/ find!

**Comment by BernieSumption** [comment]
I tried this, but all my dependencies resolved through webpack modules are still coming up as "undetermined".

Popping in there with a debugger, it seems that the issue is with the isModule function:

```js
function isModule(pDependency, pModules = ["node_modules"]) {
    return pModules.some(
        pModule => pDependency.resolved.includes(pModule)
    );
}
```

My modules folder is an absolute path, but `pDependency.resolved` is a relative path from the project root.

**Comment by BernieSumption** [comment]
Yup, working for me now :)

**Comment by sverweij** [comment]
@BernieSumption the PR is merged to the main branch and published as a regular release (v4.3.0). Thanks again for filing the issue and helping with the validation!
